File: SPCode/CS/Data/Support/SPCompanyData.cs
using System.Text.RegularExpressions;
using System.Diagnostics;
using System;
using System.Drawing.Drawing2D;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data.SqlClient;
using System.Data;
using System.Collections.Generic;
using WeifenLuo.WinFormsUI;
using Microsoft.Win32;
using WeifenLuo;
using System.IO;
using System.Drawing.Imaging;
using SoftLogik.Win.SPDataProxyTableAdapters;


namespace SoftLogik.Win
{
	public class SPCompanyData
	{
		
		public static SPDataProxy.SPCompanyDataTable GetCompanies()
		{
			SPDataProxy.SPCompanyDataTable data;
			using (SPDataProxyTableAdapters.taSPCompany dataCmd = new SPDataProxyTableAdapters.taSPCompany())
			{
				data = dataCmd.GetCompanies();
			}
			
			
			return data;
		}
		public static SPDataProxy.SPCompanyDataTable GetEmptyRecord()
		{
			SPDataProxy.SPCompanyDataTable data;
			using (SPDataProxyTableAdapters.taSPCompany dataCmd = new SPDataProxyTableAdapters.taSPCompany())
			{
				data = dataCmd.GetEmptyCompany();
			}
			
			
			return data;
		}
		
		public static int InsertCompany(string Name, string PhoneList, string EmailAddress, string Address1, string Address2, byte[] Logo, string Motto, string CustomNote)
		{
			using (SoftLogik.Win.SPDataProxyTableAdapters.taSPCompany companyAdapter = new SoftLogik.Win.SPDataProxyTableAdapters.taSPCompany())
			{
				return companyAdapter.Insert(Name, PhoneList, EmailAddress, Address1, Address2, Logo, Motto, CustomNote);
			}
			
		}
		public static int UpdateCompany(long CompanyID, string Name, string PhoneList, string EmailAddress, string Address1, string Address2, byte[] Logo, string Motto, string CustomNote)
		{
			using (SoftLogik.Win.SPDataProxyTableAdapters.taSPCompany companyAdapter = new SoftLogik.Win.SPDataProxyTableAdapters.taSPCompany())
			{
				return companyAdapter.Update(CompanyID, Name, PhoneList, EmailAddress, Address1, Address2, Logo, Motto, CustomNote);
			}
			
		}
		public static int DeleteCompany(long CompanyID)
		{
			using (SoftLogik.Win.SPDataProxyTableAdapters.taSPCompany companyAdapter = new SoftLogik.Win.SPDataProxyTableAdapters.taSPCompany())
			{
				return companyAdapter.Delete(CompanyID);
			}
			
		}
	}
	
}

File: SPCode/Data/Support/SPCompanyData.vb
Imports System.IO
Imports System.Drawing.Imaging
Imports SoftLogik.Win.SPDataProxyTableAdapters

Public Class SPCompanyData
    Public Shared Function GetCompanies() As SPDataProxy.SPCompanyDataTable
        Dim data As SPDataProxy.SPCompanyDataTable
        Using dataCmd As New SPDataProxyTableAdapters.taSPCompany
            data = dataCmd.GetCompanies()
        End Using

        Return data
    End Function
    Public Shared Function GetEmptyRecord() As SPDataProxy.SPCompanyDataTable
        Dim data As SPDataProxy.SPCompanyDataTable
        Using dataCmd As New SPDataProxyTableAdapters.taSPCompany
            data = dataCmd.GetEmptyCompany()
        End Using

        Return data
    End Function

    Public Shared Function InsertCompany(ByVal Name As String, _
    ByVal PhoneList As String, ByVal EmailAddress As String, ByVal Address1 As String, _
    ByVal Address2 As String, ByVal Logo() As Byte, ByVal Motto As String, ByVal CustomNote As String) As Integer
        Using companyAdapter As New taSPCompany
            Return companyAdapter.Insert(Name, PhoneList, EmailAddress, Address1, Address2, Logo, Motto, CustomNote)
        End Using
    End Function
    Public Shared Function UpdateCompany(ByVal CompanyID As Long, ByVal Name As String, _
    ByVal PhoneList As String, ByVal EmailAddress As String, ByVal Address1 As String, _
    ByVal Address2 As String, ByVal Logo() As Byte, ByVal Motto As String, ByVal CustomNote As String) As Integer
        Using companyAdapter As New taSPCompany
            Return companyAdapter.Update(CompanyID, Name, PhoneList, EmailAddress, Address1, Address2, Logo, Motto, CustomNote)
        End Using
    End Function
    Public Shared Function DeleteCompany(ByVal CompanyID As Long) As Integer
        Using companyAdapter As New taSPCompany
            Return companyAdapter.Delete(CompanyID)
        End Using
    End Function
End Class
